adapters/freewheelssp/freewheelssp.go

```diff
@@ -0,0 +1,119 @@
+package freewheelssp
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb/v16/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"golang.org/x/text/currency"
+	"net/http"
+)
+
+type RequestImpExt struct {
+	ZoneId int `json:"zoneId"`
+}
+
+type adapter struct {
+	endpoint string
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	if len(request.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{
+			Message: "No impression in bid request.",
+		}}
+	}
+
+	for i := 0; i < len(request.Imp); i++ {
+		imp := &request.Imp[i]
+		var bidderExt adapters.ExtImpBidder
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			return nil, []error{&errortypes.BadInput{
+				Message: fmt.Sprintf("Invalid imp.ext, %s", err.Error()),
```

Comment:
Consider adding the imp id or index to the error message to give the publisher a better idea of where the mistake is made. (both here and on line 42)

Reply:
got it